Exiled.Events/Patches/Events/Player/ChangingRole.cs

```diff
@@ -160,8 +159,7 @@ private static bool Prefix(CharacterClassManager __instance, ref RoleType classi
 
                 ListPool<Inventory.SyncItemInfo>.Shared.Return(list);
 
-                var changedRoleEventArgs1 = new ChangedRoleEventArgs(player, oldRole, startItemsList, false, escape);
-                Player.OnChangedRole(changedRoleEventArgs1);
+                Player.OnChangedRole(changedRoleEventArgs);
```

Comment:
Remember to set IsEscaped to false.
You sholdn't instantiate `ChangedRoleEventArgs` immediately.
You should instantiate a new `ChangedRoleEventArgs` here with IsEscaped set to false and above inside the `if (lite)` with IsEscaped set to true, we'll only instantiate one object anyway.

Reply:
Lite doesn't set isescaped to true. It set's has position override to true, or whatever it is. The only difference between the two initializations is that if lite is true, the it's set to true, otherwise it's set to false, which is what's being done here.

Reply:
I don't think there's a real issue with initializing it immediately. It saves 2 lines that would be used for the initialization before running the event, and it's never going to go to waste. Plus, nothing inside of the events is modified after the initialization.

Reply:
It's fine then, I misunderstood one of your previous commit.